build/tools/avalanchejs/create-an-asset-on-the-x-chain.md

```diff
@@ -1,112 +1,151 @@
 # Create an Asset on the X-Chain
 
-This example creates an asset in the X-Chain and publishes it to the Avalanche platform. The first step in this process is to create an instance of AvalancheJS connected to our Avalanche platform endpoint of choice.
-
-```text
-import {
-    Avalanche,
-    BinTools,
-    Buffer,
-    BN
-  } from "avalanche" 
-import {
-    InitialStates,
-    SECPTransferOutput
-  } from "avalanche/dist/apis/avm"
-
-let myNetworkID = 12345; //default is 3, we want to override that for our local network
-let myBlockchainID = "GJABrZ9A6UQFpwjPU8MDxDd8vuyRoDVeDAXc694wJ5t3zEkhU"; // The X-Chain blockchainID on this network
-let avax = new Avalanche("localhost", 9650, "http", myNetworkID, myBlockchainID);
-let xchain = avax.XChain(); //returns a reference to the X-Chain used by AvalancheJS
+This example creates an asset in the X-Chain and publishes it to the Avalanche platform. The first step in this process is to create an instance of AvalancheJS connected to our Avalanche platform endpoint of choice. In this example we're using the local network `12345` via [Avash](../avash). The code examples are written in typescript. The script is in full, in both typescript and javascript, after the individual steps.
+
+```ts
+import { 
+  Avalanche,
+  BinTools,
+  BN,
+  Buffer
+ } from "avalanche"
+import { 
+  AVMAPI, 
+  InitialStates, 
+  KeyChain,
+  SECPMintOutput, 
+  SECPTransferOutput, 
+  Tx,
+  UnsignedTx,
+  UTXOSet
+} from "avalanche/dist/apis/avm"
+import { 
+  iAVMUTXOResponse 
+} from "avalanche/dist/apis/avm/interfaces"
+      
+const ip: string = "localhost"
+const port: number = 9650
+const protocol: string = "http"
+const networkID: number = 12345 // Default is 1, we want to override that for our local network
+const avalanche: Avalanche = new Avalanche(ip, port, protocol, networkID)
+const xchain: AVMAPI = avalanche.XChain() // Returns a reference to the X-Chain used by AvalancheJS
+```
+
+## Import the local network's whale
+
+Next we get an instance of bintools, for dealing with binary data, an the X-Chain local keychain. The local network `12345` has a whale which you can access with the private key `PrivateKey-ewoqjP7PxY4yr3iLTpLisriqt94hdyDFNgchSxGGztUrTXtNN`. Lastly get the whale address as a `Buffer` and as a `string`.
+
+```ts
+const bintools: BinTools = BinTools.getInstance()
+const xKeychain: KeyChain = xchain.keyChain()
+const privKey: string = "PrivateKey-ewoqjP7PxY4yr3iLTpLisriqt94hdyDFNgchSxGGztUrTXtNN"
+xKeychain.importKey(privKey)
+const xAddresses: Buffer[] = xchain.keyChain().getAddresses()
+const xAddressStrings: string[] = xchain.keyChain().getAddressStrings()
+```
+
+## Prepare for the Mint Output
+
+Now we need to create an empty array for the `SECPMintOutput` which we're going to create. We also need a `threshold` and `locktime` for the outputs which we're going to create. Each X-Chain transaction can accepts a 256 byte `memo` of arbitrary data.
```

Comment:
`can accepts a 256 byte memo` --> `can contain a memo field of up to 256 bytes.`